quilt/tools/store.py

```diff
@@ -298,18 +310,29 @@ def _add_to_contents(self, fullname, objhash, ext, path, target):
         """
         contents = self.get_contents()
         ipath = fullname.split('.')
-        dfname = ipath.pop()
+        leaf = ipath.pop()
 
         ptr = contents
         for node in ipath:
             ptr = ptr.setdefault(node, {TYPE_KEY: NodeType.GROUP.value})
 
-        ptr[dfname] = dict({TYPE_KEY: NodeType.TABLE.value},
-                           hashes=[objhash],
-                           metadata=dict(q_ext=ext,
-                                         q_path=path,
-                                         q_target=target)
-                          )
+        try:
+            target_type = TargetType(target)
+            if target_type is TargetType.PANDAS:
+                node_type = NodeType.TABLE
+            elif target_type is TargetType.FILE:
+                node_type = NodeType.FILE
+            else:
+                raise NotImplementedError()
```

Comment:
Yeah; adding it to the enum but not here would be a bug in our code (rather than e.g. a user error - trying to use an unsupported target, etc.), so I think `assert` is more appropriate.